pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -169,11 +175,12 @@ type PipelineTask struct {
 	// Conditions is a list of conditions that need to be true for the task to run
 	// Conditions are deprecated, use WhenExpressions instead
 	// +optional
+	// +listType=atomic
 	Conditions []PipelineTaskCondition `json:"conditions,omitempty"`
 
-	// WhenExpressions is a list of when expressions that need to be true for the task to run
+	// When is a list of when expressions that need to be true for the task to run
 	// +optional
-	WhenExpressions WhenExpressions `json:"when,omitempty"`
+	When WhenExpressions `json:"when,omitempty"`
```

Comment:
Any reason for this change ? 🙃

Reply:
The OpenAPI rules consider the field name not matching the JSON tag to be a rule violation, so I renamed this (and the other one) to match. If we want to get to no rule violations, we will need to rename the fields eventually.

Reply:
Lol.. really ?!? That's.. interesting.. it is in the code, it doesn't "show" to the user..

Reply:
Yeah, I can only assume they want to make sure the OpenAPI field name matches the Go API field name for consistency. 🤷